File: 1_forloop_add_test/forloop_add_test.pl
#!/usr/bin/env perl

use strict;

sub main {
    my ($toval) = @_;
    my $sumval = 0;
    for (my $i=0; $i < $toval; $i++) {
        for (my $j=0; $j < $toval; $j++) {
            $sumval += $i+($j^1);
        }
    }
    print 'Perl: ', $sumval, "\n";
}

main ( int($ARGV[0]) );

File: 1_forloop_add_test/forloop_add_test.js
#!/usr/bin/env nodejs

function main(toval) {
    var i, j, sumval = 0;
    for (i=0; i<toval; i++) {
        for (j=0; j<toval; j++) {
            sumval += i+(j^1);
        }
    }
    console.log("nodejs:", sumval);
}

main( parseInt(process.argv.slice(2)) );
